src/main/fc/rc.c

```diff
@@ -489,141 +491,177 @@ static FAST_CODE void processRcSmoothingFilter(void)
-    setpointAcceleration = prevAcceleration[axis] + pid->feedforwardSmoothFactor * (setpointAcceleration - prevAcceleration[axis]);
-    prevAcceleration[axis] = setpointAcceleration * zeroTheAcceleration;
-    setpointAcceleration = setpointAcceleration * pid->feedforwardBoostFactor * jitterAttenuator * zeroTheAcceleration;
+    // smooth the setpointDelta element (effectively a second order filter since incoming setpoint was already smoothed)
+    setpointSpeedDelta = prevSetpointSpeedDelta[axis] + pid->feedforwardSmoothFactor * (setpointSpeedDelta - prevSetpointSpeedDelta[axis]);
+    prevSetpointSpeedDelta[axis] = setpointSpeedDelta;
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 1, lrintf(setpointSpeed * 0.1f)); // base feedforward without acceleration
-    }
+    // apply gain factor to delta and adjust for rxRate
+    const float feedforwardBoost = setpointSpeedDelta * rxRate * pid->feedforwardBoostFactor;
 
-    float feedforward = setpointSpeed + setpointAcceleration;
+    feedforward = setpointSpeed;
 
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 2, lrintf(feedforward * 0.1f));
-        // un-smoothed feedforward including acceleration but before limiting, transition, averaging, and jitter reduction
-    }
+    if (axis < FD_YAW) {
+        // for pitch and roll, add setpointSpeedDelta element (boost)
+        feedforward += feedforwardBoost;
+        // apply jitter reduction multiplier
+        feedforward *= jitterAttenuator;
+        // pull feedforward back towards zero as sticks approach max if in same direction
+        if (pid->feedforwardMaxRateLimit && feedforward * setpoint > 0.0f) {
+            const float limit = (maxRcRate[axis] - fabsf(setpoint)) * pid->feedforwardMaxRateLimit;
+            feedforward = (limit > 0.0f) ? constrainf(feedforward, -limit, limit) : 0.0f;
+        }
 
-    // apply feedforward transition
+    } else {
+        // for yaw, only apply jitter reduction only to the feedforward delta element
+        // can't be applied to the 'sustained' element since it will divide it to a small fraction of what is needed
+        feedforward *= jitterAttenuator;
+
+        // instead of adding setpoint acceleration, which is too aggressive,
+        // add a slow-fading high-pass filtered setpoint element
+        // this provides a 'sustained boost' with low noise
+        // it mimics the normal sustained yaw motor drive requirements, reducing P and I and hence reducing bounceback
+        // this doesn't add significant noise to feedforward
+        // too little yaw FF causes iTerm windup and slow bounce back when stopping a hard yaw
+        // too much causes fast bounce back when stopping a hard yaw
+
+        // calculate lowpass gain factor from user specified time constant, can't exceed 1.0
+        const float gain = pt1FilterGainFromDelay(pid->feedforwardYawHoldTime, rxInterval);
+        // subtract lowpass from input to get highpass
```

Comment:
This comment is meant to be in line 627, I think

Reply:
thanks and I tidied up the related comments.